corehq/apps/userreports/dbaccessors.py

```diff
@@ -65,3 +70,30 @@ def get_all_report_configs():
 def delete_all_report_configs():
     from corehq.apps.userreports.models import ReportConfiguration
     delete_all_docs_by_doc_type(ReportConfiguration.get_db(), ('ReportConfiguration',))
+
+
+def get_all_table_names_for_domain(domain):
+    engine = connection_manager.get_engine(UCR_ENGINE_ID)
+    with engine.begin() as connection:
+        possible_matches = connection.execute("select tablename from pg_tables where tablename ~ %s", domain)
+    # DO NOT assume just because the domain is in the table name
+    # that the table is in the domain
+    all_table_names_for_domain = []
+    for table_name, in possible_matches:
+        match = re.match(r'^(config_report|ucr)_(.*)_([0-9a-f]{0,29})_([0-9a-f]{8})$', table_name)
```

Comment:
Not sure how much performance matters here, but might make sense to compile this regex outside the loop.